graylog2-web-interface/src/stores/inputs/InputStatesStore.js

```diff
@@ -28,13 +28,53 @@ const InputStatesStore = Reflux.createStore({
             }
             result[input.id][node] = input;
           });
-        })
+        });
         this.inputStates = result;
         this.trigger({inputStates: this.inputStates});
 
         return result;
       });
-  }
+  },
+
+  start(input) {
+    let url;
+
+    if (input.global) {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.ClusterInputStatesController.start(input.id).url);
+    } else {
+      url = URLUtils.qualifyUrl(jsRoutes.controllers.api.InputStatesController.start(input.id).url);
+    }
+
+    return fetch('PUT', url)
+      .then(
+        () => {
+          this.list();
+          UserNotification.success(`Input '${input.title}' started successfully`);
```

Comment:
This does not really take the response into consideration, it just checks if the server response is a 200, which is the case even if starting the input failed on all nodes concerned.

Maybe we should add a simple check for the presence of any null elements in the hash response. If there are successful (!= null) and failed (== null) entries, we could notify a warning, if there are only null entries we should indicate an error.